desc/grid.py

```diff
@@ -598,6 +598,52 @@ def replace_at_axis(self, x, y, copy=False, **kwargs):
             )
         return x
 
+    def meshgrid_reshape(self, x, order):
+        """Reshape data to match grid coordinates.
+
+        Given flattened data on a tensor product grid, reshape the data such that
+        the axes of the array correspond to coordinate values on the grid.
+
+        Parameters
+        ----------
+        x : ndarray, shape(N,) or shape(N,3)
+            Data to reshape.
+        order : str
+            Desired order of axes for returned data. Should be a permutation of
+            ``grid.coordinates``, eg ``order="rtz"`` has the first axis of the returned
+            data correspond to different rho coordinates, the second axis to different
+            theta, etc.  ``order="trz"`` would have the first axis correspond to theta,
+            and so on.
+
+        Returns
+        -------
+        x : ndarray
+            Data reshaped to align with grid nodes.
+        """
+        errorif(
+            not self.is_meshgrid,
+            ValueError,
+            "grid is not a tensor product grid, so meshgrid_reshape doesn't "
+            "make any sense",
+        )
+        errorif(
+            sorted(order) != sorted(self.coordinates),
+            ValueError,
+            f"order should be a permutation of {self.coordinates}, got {order}",
+        )
+        shape = (self.num_poloidal, self.num_rho, self.num_zeta)
+        vec = False
+        if x.ndim > 1:
+            vec = True
+            shape += (-1,)
+        x = x.reshape(shape, order="F")
```

Comment:
`f = grid.meshgrid_reshape(f)` seems simpler than `f = f.reshape(grid.num_poloidal, grid.num_radial, grid.num_toroidal, order="F").`

Is it worth to default `order=None` and early exit?

```python
inorder = self.coordinates[1] + self.coordinates[0] + self.coordinates[2]
order = setdefault(order, inorder)
...
x = x.reshape(shape, order="F")
if order == inorder:
  return x
x = jnp.swapaxes(x, 1, 0)
```

Reply:
I considered that but thought it would be clearer and less error prone to make the ordering explicit.